tests/components/test_gpio_pin.py

```diff
@@ -315,3 +315,21 @@ def test_analogue_value_setter() -> None:
 
     with pytest.raises(ValueError):
         pin.analogue_value = -1
+
+
+class Peripheral(DerivedComponent):
+    """A mock derived component."""
+
+
+def test_derived_mode_is_possible() -> None:
+    """Check that it is possible to support a derived component."""
+    driver = MockGPIOPinDriver()
+    GPIOPin(
+        0,
+        driver,
+        initial_mode=GPIOPinMode.ANALOGUE_OUTPUT,
+        hardware_modes={
+            GPIOPinMode.ANALOGUE_OUTPUT,
+            GPIOPinMode.PWM_OUTPUT,
+        },
```

Comment:
I think there should be a `firmware_modes = {Peripheral}` here.

Reply:
oops

dc1d748